app/Console/Commands/DailyChallengeUserStatsRecalculate.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Console\Commands;
+
+use App\Models\DailyChallengeUserStats;
+use Illuminate\Console\Command;
+
+class DailyChallengeUserStatsRecalculate extends Command
+{
+    protected $signature = 'daily-challenge:user-stats-recalculate {--all} {id?*}';
+
+    protected $description = 'Recalculate user daily challenge stats';
+
+    public function handle(): int
+    {
+        $isAll = $this->option('all');
+        $userIds = $this->argument('id');
+
+        if (count($userIds) > 0 && $isAll) {
```

Comment:
Also something to differentiate `--all` with no changes and no options passed? They look the same right now.